test/TEMPFILE/TEMPFILEENCODING.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+#
+# MIT License
+#
+# Copyright The SCons Foundation
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY
+# KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+"""
+Verify that setting the $TEMPFILEENCODING variable will be used for
+encoding the file contents when writing the generated tempfile used
+for long command lines.
+"""
+
+
+import TestSCons
+
+test = TestSCons.TestSCons()
+
+SCONSTRUCT_TEMPLATE="""
+import SCons.Platform
+
+command = 'xyz ./test€/file.c'
+encoding = {encoding!r}
+
+tempfileencoding = {tempfileencoding}
+defaultencoding = {defaultencoding}
+
+DefaultEnvironment()
```

Comment:
Can we add `tools=[]` here? It will speed up the test by avoiding tool initializations..

Reply:
Yes.  That is a bug.